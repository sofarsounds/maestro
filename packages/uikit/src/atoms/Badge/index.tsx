import styled, { css } from '../../lib/styledComponents';

interface BadgeProps {
  children: any;
  colour?: 'primary';
  clickable?: boolean;
  disabled?: boolean;
}

const Badge = styled.div<BadgeProps>`
  ${({ theme, colour, clickable }) => css`
    width: 36px;
    height: 36px;
    font-size: ${theme.fontSizes.body2};
    transition: ${theme.utils.transition()};
    border: none;
    border-radius: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    outline: none;
    user-select: none;

    ${clickable &&
      css`
        cursor: pointer;

        &:hover {
          background-color: rgba(17, 17, 17, 0.1);
        }
      `}

    ${colour === 'primary' &&
      css`
        background-color: ${theme.colours.primary};
        color: ${theme.colours.whiteDenim};
        font-weight: 600;

        ${clickable &&
          css`
            &:hover {
              background-color: ${theme.colours.green800};
            }
          `}
      `}
  `}
`;

export default Badge;
