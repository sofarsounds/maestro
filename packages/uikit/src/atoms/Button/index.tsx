import styled, { keyframes, css } from 'styled-components';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

interface ButtonProps {
  loading?: boolean;
}

export const BaseButton = styled.button<ButtonProps>`
  ${({ theme, loading }) => css`
    border-radius: ${theme.borderRadius.button};

    cursor: pointer;
    font-size: ${theme.fontSizes.button};
    font-weight: 600;
    height: 44px;
    letter-spacing: 0.8px;
    outline: none;
    padding: 0 25px;
    text-decoration: none;
    text-transform: uppercase;
    transition: ${theme.utils.transition()};

    &:disabled {
      cursor: not-allowed;
      pointer-events: none;
    }

    ${loading &&
      css`
        pointer-events: none;
        cursor: disabled;
      `}
  `}
`;

export const PrimaryButton = styled(BaseButton)`
  ${({ theme, loading }) => css`
    background: ${theme.colours.primary};
    border: 1px solid ${theme.colours.primary};
    color: ${theme.colours.whiteDenim};

    &:hover {
      background-color: ${theme.colours.green900};
    }

    &:focus {
      background-color: ${theme.colours.green700};
    }

    &:disabled {
      background-color: ${theme.colours.blueSmoke};
      border-color: ${theme.colours.blueSmoke};
    }

    ${loading &&
      css`
        color: transparent !important;
        background-color: ${theme.colours.blueSmoke};
        border-color: ${theme.colours.blueSmoke};
        position: relative;

        :after {
          animation: ${rotate360} 0.5s linear infinite;
          border: 0.1rem solid ${theme.colours.whiteDenim};
          border-radius: 50%;
          border-right-color: transparent;
          border-top-color: transparent;
          content: '';
          display: block;
          height: 0.8rem;
          left: 50%;
          margin-left: -0.4rem;
          margin-top: -0.4rem;
          position: absolute;
          top: 50%;
          width: 0.8rem;
          z-index: 1;
        }
      `};
  `}
`;

export const SecondaryButton = styled(PrimaryButton)`
  ${({ theme, loading }) => css`
    background: ${theme.colours.whiteDenim};
    color: ${theme.colours.primary};

    &:hover {
      border-color: ${theme.colours.green900};
      color: ${theme.colours.green900};
      background-color: ${theme.colours.whiteDenim};
    }

    &:focus {
      background-color: ${theme.colours.green50};
    }

    &:disabled {
      background-color: ${theme.colours.whiteDenim};
      color: ${theme.colours.blueSmoke};
    }

    ${loading &&
      css`
        :after {
          border-color: ${theme.colours.blueSmoke};
          border-right-color: transparent;
          border-top-color: transparent;
        }
      `}
  `}
`;

export const TertiaryButton = styled(SecondaryButton)`
  ${({ theme, loading }) => css`
    background: ${theme.colours.whiteDenim};
    border-color: ${theme.colours.backToBlack};
    color: ${theme.colours.backToBlack};

    &:hover {
      border-color: ${theme.colours.black};
      color: ${theme.colours.black};
      background-color: ${theme.colours.whiteDenim};
    }

    &:focus {
      background-color: ${theme.colours.macyGrey};
    }

    ${loading &&
      css`
        border-color: ${theme.colours.blueSmoke};

        :after {
          border-color: ${theme.colours.blueSmoke};
          border-right-color: transparent;
          border-top-color: transparent;
        }
      `}
  `}
`;

export const LinkButton = styled(SecondaryButton)`
  ${({ theme }) => css`
    background: ${theme.colours.whiteDenim};
    border-color: ${theme.colours.whiteDenim};
    color: ${theme.colours.primary};

    &:hover {
      color: ${theme.colours.green900};
      border-color: ${theme.colours.whiteDenim};
    }

    &:focus {
      background-color: ${theme.colours.green50};
      border-color: ${theme.colours.green50};
    }

    &:disabled {
      background-color: ${theme.colours.whiteDenim};
      border-color: ${theme.colours.whiteDenim};
    }
  `};
`;
