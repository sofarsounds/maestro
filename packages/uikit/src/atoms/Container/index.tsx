import styled, { css } from 'styled-components';

type Sizes = 'default' | 'small' | 'large' | 'fullscreen';
interface ContainerProps {
  size?: string;
}

export default styled.div<ContainerProps>`
  ${({ theme, size }) => css`
    width: 100%;
    max-width: ${theme.dimensions.containerWidth[(size || 'default') as Sizes]};
    margin: 0 auto;
  `}
`;
