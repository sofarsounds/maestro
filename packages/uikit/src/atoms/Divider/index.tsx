import styled from 'styled-components';

export default styled.hr`
  border: none;
  border-top: 1px solid ${({ theme }) => theme.colours.macyGrey};
  width: 100%;
  margin: ${({ theme }) => theme.ruler[11]}px 0;
`;
