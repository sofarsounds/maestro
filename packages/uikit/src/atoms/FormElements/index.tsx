export { default as Textfield } from './Textfield';
export { default as Textarea } from './Textarea';
export { default as Select } from './Select';
