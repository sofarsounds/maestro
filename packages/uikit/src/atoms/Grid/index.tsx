import styled, { css } from 'styled-components';

/**
 * TODO
 * - Auto-stretch columns to fit when no width is given
 *
 */

interface ColProps {
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
}
export const Col = styled.div<ColProps>`
  ${({ theme, xs, sm, md, lg }) => css`
    grid-column-end: span ${xs || 1};

    ${sm &&
      theme.media.tablet`
      grid-column-end: span ${sm};
    `}

    ${md &&
      theme.media.laptop`
      grid-column-end: span ${md};
    `}

    ${lg &&
      theme.media.desktop`
      grid-column-end: span ${lg};
    `}
  `}
`;

// Grid
interface GridProps {
  cols?: number;
  gap?: number;
}
export default styled.div<GridProps>`
  ${({ cols, gap }) => css`
    display: grid;
    grid-template-columns: repeat(${cols || 12}, 1fr);
    grid-gap: ${gap || '20px'};
  `}
`;
