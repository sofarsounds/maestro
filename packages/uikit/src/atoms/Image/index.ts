import styled, { css } from 'styled-components';

export default styled.img`
  ${({ theme }) => css`
    border-radius: ${theme.borderRadius.default};
    width: 100%;
  `}
`;
