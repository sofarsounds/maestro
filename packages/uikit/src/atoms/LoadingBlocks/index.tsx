import styled, { css, keyframes } from 'styled-components';

const animation = keyframes`
    0%{
        background-position: 0px 0
    }
    100%{
        background-position: 4000px 0
    }
}
`;

interface BoxProps {
  width: string;
}

interface RectangleProps {
  width: string;
  height: string;
}

const Animation = styled.div`
  animation-duration: 1.25s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: ${animation};
  animation-timing-function: linear;
  background: #f6f6f6;
  background: linear-gradient(to right, #f6f6f6 8%, #f2f2f2 18%, #f6f6f6 33%);
  background-size: 4000px 104px;
  position: relative;
`;

const Box = styled(Animation)<BoxProps>`
  ${({ width }) => css`
    width: ${width};
    height: ${width};
    border-radius: 2px;
  `}
`;

const Circle = styled(Animation)<BoxProps>`
  ${({ width }) => css`
    width: ${width};
    height: ${width};
    border-radius: 100%;
  `}
`;

const Rectangle = styled(Animation)<RectangleProps>`
  ${({ width, height }) => css`
    width: ${width};
    height: ${height};
    border-radius: 2px;
  `}
`;

export default {
  Box,
  Circle,
  Rectangle
};
