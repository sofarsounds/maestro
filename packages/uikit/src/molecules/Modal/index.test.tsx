import React from 'react';
import { ReactWrapper } from 'enzyme';
import { mountWithTheme } from '../../test';

import { Modal } from './';
import { LinkButton as Button } from '../../atoms/Button';

let wrapper: ReactWrapper;

const hideMock = jest.fn();

function setup(isShowing: boolean, hide: () => {}, children: any) {
  return mountWithTheme(
    <Modal isShowing={isShowing} hide={hide}>
      {children}
    </Modal>
  );
}

describe('<Modal />', () => {
  describe('When isShowing is true', () => {
    beforeEach(() => {
      wrapper = setup(true, hideMock, <>Some children</>);
    });

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Exiting the modal when it is open', () => {
    beforeEach(() => {
      wrapper = setup(true, hideMock, <>Some visible text</>);
    });

    it('closes the modal when you click on the X button', () => {
      const xButton = wrapper.find(Button);

      xButton.simulate('click');

      expect(hideMock).toHaveBeenCalledTimes(1);
    });
  });

  describe('When isShowing is false', () => {
    beforeEach(() => {
      wrapper = setup(false, hideMock, <>Some other children</>);
    });

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('does not render anything', () => {
      expect(wrapper).toEqual({});
    });
  });
});
