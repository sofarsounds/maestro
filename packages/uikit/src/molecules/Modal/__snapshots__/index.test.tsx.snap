// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> When isShowing is false matches the snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#039842",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "54px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <Component
    hide={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    isShowing={false}
  />
</ThemeProvider>
`;

exports[`<Modal /> When isShowing is true matches the snapshot 1`] = `
.c4 {
  border-radius: 4px;
  cursor: pointer;
  font-size: 14px;
  font-weight: 600;
  height: 44px;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
  outline: none;
  padding: 0 25px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background: #039842;
  border: 1px solid #039842;
  color: #FFFFFF;
  background: #FFFFFF;
  color: #039842;
  background: #FFFFFF;
  border-color: #FFFFFF;
  color: #039842;
}

.c4:disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.c4:hover {
  background-color: #006721;
}

.c4:focus {
  background-color: #039842;
}

.c4:disabled {
  background-color: #949494;
  border-color: #949494;
}

.c4:hover {
  border-color: #006721;
  color: #006721;
  background-color: #FFFFFF;
}

.c4:focus {
  background-color: #E6F7EA;
}

.c4:disabled {
  background-color: #FFFFFF;
  color: #949494;
}

.c4:hover {
  color: #006721;
  border-color: #FFFFFF;
}

.c4:focus {
  background-color: #E6F7EA;
  border-color: #E6F7EA;
}

.c4:disabled {
  background-color: #FFFFFF;
  border-color: #FFFFFF;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1040;
  opacity: 0.5;
  background-color: #000;
}

.c1 {
  display: inline-table;
  position: fixed;
  padding: 0.5rem;
  top: 0;
  left: 0;
  z-index: 1050;
  width: 100%;
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  outline: 0;
}

.c2 {
  z-index: 100;
  background: white;
  margin: 1.75rem 1rem;
  position: fixed;
  left: 0;
  top: 0;
  margin: 0;
  width: 100%;
  height: 100vh;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c5 {
  pointer-events: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem 1rem;
  min-height: 250px;
}

@media (min-width:768px) {
  .c2 {
    border-radius: 3px;
    position: relative;
    margin: 1.75rem auto;
    width: 100%;
    left: auto;
    top: auto;
    min-height: 200px;
    height: auto;
  }
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#039842",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "54px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <Component
    hide={[MockFunction]}
    isShowing={true}
  >
    <Portal
      containerInfo={
        .c4 {
  border-radius: 4px;
  cursor: pointer;
  font-size: 14px;
  font-weight: 600;
  height: 44px;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
  outline: none;
  padding: 0 25px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background: #039842;
  border: 1px solid #039842;
  color: #FFFFFF;
  background: #FFFFFF;
  color: #039842;
  background: #FFFFFF;
  border-color: #FFFFFF;
  color: #039842;
}

.c4:disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.c4:hover {
  background-color: #006721;
}

.c4:focus {
  background-color: #039842;
}

.c4:disabled {
  background-color: #949494;
  border-color: #949494;
}

.c4:hover {
  border-color: #006721;
  color: #006721;
  background-color: #FFFFFF;
}

.c4:focus {
  background-color: #E6F7EA;
}

.c4:disabled {
  background-color: #FFFFFF;
  color: #949494;
}

.c4:hover {
  color: #006721;
  border-color: #FFFFFF;
}

.c4:focus {
  background-color: #E6F7EA;
  border-color: #E6F7EA;
}

.c4:disabled {
  background-color: #FFFFFF;
  border-color: #FFFFFF;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1040;
  opacity: 0.5;
  background-color: #000;
}

.c1 {
  display: inline-table;
  position: fixed;
  padding: 0.5rem;
  top: 0;
  left: 0;
  z-index: 1050;
  width: 100%;
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  outline: 0;
}

.c2 {
  z-index: 100;
  background: white;
  margin: 1.75rem 1rem;
  position: fixed;
  left: 0;
  top: 0;
  margin: 0;
  width: 100%;
  height: 100vh;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c5 {
  pointer-events: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem 1rem;
  min-height: 250px;
}

@media (min-width:768px) {
  .c2 {
    border-radius: 3px;
    position: relative;
    margin: 1.75rem auto;
    width: 100%;
    left: auto;
    top: auto;
    min-height: 200px;
    height: auto;
  }
}

<body>
          <div
            class="c0"
          />
          <div
            class="c1"
          >
            <div
              class="c2"
            >
              <div
                class="c3"
              >
                <button
                  class="c4"
                >
                  X
                </button>
              </div>
              <div
                class="c5"
              >
                Some children
              </div>
            </div>
          </div>
        </body>
      }
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ksYbfQ",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1040;
  opacity: 0.5;
  background-color: #000;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ksYbfQ",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          />
        </StyledComponent>
      </styled.div>
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-hmzhuo",
                "isStatic": true,
                "lastClassName": "c1",
                "rules": Array [
                  "
  display: inline-table;
  position: fixed;
  padding: 0.5rem;
  top: 0;
  left: 0;
  z-index: 1050;
  width: 100%;
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  outline: 0;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-hmzhuo",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c1"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-frDJqD",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "
  ",
                        [Function],
                        "
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-frDJqD",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="c2"
                >
                  <styled.div>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-kvZOFW",
                            "isStatic": true,
                            "lastClassName": "c3",
                            "rules": Array [
                              "
  display: flex;
  width: 100%;
  justify-content: flex-end;
",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-kvZOFW",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c3"
                      >
                        <Styled(Styled(Styled(styled.button)))
                          onClick={[Function]}
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-ifAKCX",
                                  "isStatic": false,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    "
  ",
                                    [Function],
                                    "
",
                                    "
  ",
                                    [Function],
                                    "
",
                                    "
  ",
                                    [Function],
                                    "
",
                                    "
  ",
                                    [Function],
                                    ";
",
                                  ],
                                },
                                "displayName": "Styled(Styled(Styled(styled.button)))",
                                "foldedComponentIds": Array [
                                  "sc-bwzfXH",
                                  "sc-htpNat",
                                  "sc-bxivhb",
                                ],
                                "render": [Function],
                                "styledComponentId": "sc-ifAKCX",
                                "target": "button",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            onClick={[Function]}
                          >
                            <button
                              className="c4"
                              onClick={[Function]}
                            >
                              X
                            </button>
                          </StyledComponent>
                        </Styled(Styled(Styled(styled.button)))>
                      </div>
                    </StyledComponent>
                  </styled.div>
                  <styled.div>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-hqyNC",
                            "isStatic": true,
                            "lastClassName": "c5",
                            "rules": Array [
                              "
  pointer-events: auto;
  display: flex;
  padding: 0.5rem 1rem;
  min-height: 250px;
",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-hqyNC",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c5"
                      >
                        Some children
                      </div>
                    </StyledComponent>
                  </styled.div>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </Portal>
  </Component>
</ThemeProvider>
`;
